### PR TITLE
3d tiles

### LINKED ISSUE #3940: 3D does not work in EDGE or IE9
open in win10 edge BUraut

https://mf-geoadmin3.int.bgdi.ch/3dtiles/index.html?lang=en&topic=ech&tileset3d=ch.swisstopo.swissnames3d.3d,ch.swisstopo.swisstlm3d.3d,ch.swisstopo.vegetation.3d&tilesetTs=20170814,20170425,20170630_2000_1000,20161121&bgLayer=ch.swisstopo.pixelkarte-farbe&dev3d=true&layers=WMS%7C%7Ctlm3d%20wms%20dev%7C%7Chttp:%2F%2Fwms-bod.dev.bgdi.ch%2F%3F%7C%7Cch.swisstopo.swisstlm3d-karte-farbe.3d%7C%7C1.3.0,ch.swisstopo.swissimage-product&layers_visibility=false,false&lon=7.41969&lat=46.95438&elevation=736&heading=34.892&pitch=0.278

### PR DESCRIPTION
**Pull request by oterral**
Add generic 3d tileset managment for future tilesets vegetation and swissnames
Fix #3940

[Test](//mf-geoadmin3.int.bgdi.ch/3dtiles/index.html)

### CODE REVIEW

**Comment by oterral** [comment]
jenkins test

**Comment by oterral** [comment]
jenkins test

**Comment by loicgasser** [comment]
[1A[2KPhantomJS 2.1.1 (Linux 0.0.0) ga_urlutils_service gaUrlUtils #shorten() "after each" hook for "handle service error displaying a log an returning the initial url" FAILED
	Deferred tasks to flush (1): {id: 6, time: 8000}
	verifyNoPendingTasks@test/lib/angular-mocks.js:2157:60
	test/specs/UrlUtilsService.spec.js:174:38

**Comment by oterral** [comment]
Could be great to have tese 2 improvments before merging: 
https://github.com/AnalyticalGraphicsInc/cesium/pull/5770 and https://github.com/AnalyticalGraphicsInc/cesium/pull/5788

File: src/components/map/MapService.js [hunk 1]
```diff
@@ -552,13 +552,13 @@ goog.require('ga_urlutils_service');
...
-                'ch.swisstopo.bridgestest.3d',
-                'ch.swisstopo.swissnames3d.3d'
+                'ch.swisstopo.swissnames3d.3d',
+                'ch.swisstopo.vegetation.3d'
               ];
               var tilesetTs = [
-                '20161217',
-                '20161121',
-                '20160909'
+                '20170425',
```
Comment (loicgasser) [inline src/components/map/MapService.js]: please don't forget to update the codepen and doc in chsdi.

File: src/js/GaCesium.js [hunk 2]
```diff
@@ -12,14 +12,16 @@ goog.provide('ga_cesium');
...
  * @constructor
  */
 // eslint-disable-next-line no-unused-vars
 var GaCesium = function(map, gaPermalink, gaLayers, gaGlobalOptions,
-    gaBrowserSniffer, $q, $translate, $rootScope, gaBackground) {
+    gaBrowserSniffer, $q, $translate, $rootScope, gaBackground,
+    gaStyleFactory) {
   // Url of olcesium library
   var olCesiumLibUrl = gaGlobalOptions.resourceUrl;
+  console.log(olCesiumLibUrl);
```
Comment (loicgasser) [inline src/js/GaCesium.js]: console

**Comment by loicgasser** [comment]
@oterral FYI: both PRs you mentionned have been merged.

**Comment by oterral** [comment]
Yes but https://github.com/AnalyticalGraphicsInc/cesium/pull/5770 has introduce a bug. I need to check where.

**Comment by oterral** [comment]
Names added in layer manager in 3d:
[Test](https://mf-geoadmin3.int.bgdi.ch/3dtiles/index.html?lang=fr&topic=ech&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege&layers_visibility=false,false,false,false&layers_timestamp=18641231,,,&X=183215.32&Y=614663.73&zoom=8)

**Comment by davidoesch** [comment]
Not in mobile

**Comment by davidoesch** [comment]
Sorry I mean: hardly usable (with and without names) on ios11 iphone7 chrome. Serious lag compared to prod)

**Comment by loicgasser** [comment]
On 3g samsung and chrome it s pretty good on my side on mobile. Samsung galaxy s8 rules

**Comment by davidoesch** [comment]
I suspect ios11 .... has someone an ios10 available

**Comment by davidoesch** [comment]
Confirmed by moc: Not functional on ios11 iphone7

**Comment by gjn** [comment]
I've alerted Tom.

**Comment by gjn** [comment]
@davidoesch  Anyhow: is the name behaviour ok as it is? If yes, we can give it to Topo for Abnahme.

**Comment by gjn** [comment]
@cedricmoullet @davidoesch Can you update your ios11? It works fine on Tom's ios11.

**Comment by davidoesch** [comment]
Name behaviour is ok. I will organize acceptance and komm mmgt with Booth involved Departements

**Comment by oterral** [comment]
swissnames and trees has been removed from this PR.

**Pull request merged**